libdnf5/repo/repo_sack.cpp

```diff
@@ -400,28 +400,31 @@ void RepoSack::Impl::update_and_load_repos(libdnf5::repo::RepoQuery & repos, boo
 
     auto handle_repo_exception = [&](const Repo * repo, std::exception_ptr ep, bool report_key_err) {
         // Use an exception_ptr to preserve the original type of the exception, in case we re-throw it.
-        std::exception e;
+        std::exception exception;
         try {
             std::rethrow_exception(ep);
-        } catch (const std::exception & exception) {
-            e = exception;
-        }
-        if (report_key_err) {
-            try {
-                std::rethrow_if_nested(e);
-            } catch (const LibrepoError & lr_err) {
-                if (lr_err.get_code() == LRE_BADGPG) {
-                    return true;
+        } catch (const RepoDownloadError & rd_err) {
+            exception = rd_err;
+            if (report_key_err) {
+                try {
+                    std::rethrow_if_nested(rd_err);
```

Comment:
I think there is now a different bug when the `handle_repo_exception` gets non-nested `RepoDownloadError`.
For example from here: https://github.com/rpm-software-management/dnf5/blob/main/libdnf5/repo/repo_downloader.cpp#L474-L475 so basically https://github.com/rpm-software-management/dnf5/issues/2064.

It produces logs like:
```
2025-03-18T13:17:01+0000 [40529] WARNING Error loading repo "empty" (skipping due to "skip_if_unavailable=true"):
2025-03-18T13:17:01+0000 [40529] WARNING  std::exception
```
I think it can also happen from: https://github.com/rpm-software-management/dnf5/blob/main/libdnf5/repo/librepo.cpp#L117-L119

However these seem like far less serious problems so if you don't want to deal with that now I guess we can merge this and fix that later.

Reply:
Actually the issue was already present before.
Lets fix it later.

Reply:
I think its not caused by the non-nested `RepoDownloadError` but rather by `format(...)`ing the sliced `std::exception`.

Reply:
Yes, was just fixing that, it can wait for another PR.